enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,641 @@
+  - Remote storage of the backup in a service provider (S3, Azure...).
+  - Restoration of the backup in another Hosted Cluster using the already existent API.
+
+### Non-Goals
+
+- Provide a backup/restoration solution for a hosted cluster's volumes
+  - This is not covering the Volumes migrations
+
+
+### Pros and Cons
+#### Pros
+
+**Visibility:** Having an API could enhance visibility within the HostedCluster (HC) object, which serves as the primary interface for customers to manage backups. In the event of a backup failure, in addition to log entries, it could update the status.conditions within the HostedCluster object.
+
+**Integration:** Integration with the product is strengthened, meaning that adding new features becomes more straightforward. Employing tightly coupled methods could lead to poor practices.
+
+**Alerting:** The controller can notify users through various channels:
+
+- HC status.conditions
+- Log entries
+- Events
+- Alerting
+- Metrics
+
+**Validations:** In this scenario, we incorporate an additional layer of validations (CEL), managed by api-machinery. This layer promptly detects any discrepancies with the API consensus and triggers early failure. Furthermore, additional validations can be employed within the controller.
+
+**Reusability:** Leveraging the existing ETCD backup API developed by the ETCD team, we will integrate the entire API into a new structure within our ETCD API field. Within the same field, we will incorporate the necessary functionality to upload binaries to remote storage such as S3 or Azure.
+
+#### Cons
+
+**Supportability:** Development of the API and support of it during the product life-cycle.
+**Complexity:** The Hosted Control Planes team will take the responsibility to stay tuned over the API changes of the ETCD Backup.
+
+
+## Proposal
+
+The proposal commences with the acknowledgement that a user requires an ETCD backup.
+
+The primary requirement is that the ETCD `managementType` must be configured as `managed`.
+
+This backup operation will be specified within the `hostedcluster.spec.etcd.managed` API custom field under a new field named `backup`. This new field will encompass two childs, with the primary one designated for the backup process itself, directly imported from the original ETCD Backup API. Comprehensive details regarding this API can be found in the official ETCD proposal and its corresponding implementation documentation.
+
+- [ETCD Operator Backup API Design Doc](https://docs.google.com/document/d/1xBnK3D7MCblFgxQ8RUZFyWNIU_bnSFHciLhhHAeP0Xk/edit)
+- [ETCD Backup API implementation](https://github.com/openshift/api/pull/1482)
+- [Periodic Backup Controller](https://github.com/openshift/cluster-etcd-operator/pull/1074)
+
+The other field is called `storeEndpoint` which basically will take care of the export of the backup object to an external endpoint. This endpoint could be S3, Azure, etc.... Inside of this `storeEndpoint` we have some fields:
```

Comment:
Naively I would expect a field like `storeEndpoint` to hold a URI or similar - this field holds configuration for off-cluster storage, though, so, perhaps we can call it `externalStorage`?

Reply:
I've been discussing this with Adel already. For now I think we can keep this, and then modify the names (if make sense) in the final implementation.

Reply:
I agree with changing the name now or later. I'm sure we can get a more accurate name